driver-scala/src/test/scala/org/mongodb/scala/internal/FlatMapObservableTest.scala

```diff
@@ -0,0 +1,38 @@
+package org.mongodb.scala.internal
+
+import org.mongodb.scala.{ BaseSpec, Observable, Observer }
+import org.scalatest.concurrent.{ Eventually, Futures }
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise }
+
+class FlatMapObservableTest extends BaseSpec with Futures with Eventually {
+  "FlatMapObservable" should "only complete once" in {
+    val p = Promise[Unit]()
+    val completedCounter = new AtomicInteger(0)
+    Observable(1 to 100)
+      .flatMap(
+        x =>
+          (observer: Observer[_ >: Int]) => {
+            Future(()).onComplete(_ => {
+              observer.onNext(x)
+              observer.onComplete()
+            })
+          }
+      )
+      .subscribe(
+        _ => (),
+        p.failure,
+        () => {
+          completedCounter.incrementAndGet()
+          Thread.sleep(100)
+          p.trySuccess(())
```

Comment:
without this change this would fail here because there would be 2 calls to the "root" `onComplete`, which would mean we can't reliably block until the observable is exhausted if done in this fashion (deriving `Future` from `Observable`). Also, this would fail if we did `p.success(())` instead of `trySuccess` because we don't expect to get repeated `onComplete` events.